### ISSUE TITLE
series with n=None now raises StopIteration 

### ISSUE THREAD

**Issue by spj101** [sympy/sympy#20551]
Prior to commit 8a06c07 by @sachin-4099 (see pull request #19369) the series `exp(x)/x` could be lazily computed, now it instead raises `StopIteration`.

**Input**
```python
from sympy import *
x = Symbol('x')
expr = (exp(x)/x).series(x,n=None)
terms = [ next(expr) for i in range(2) ]
print(terms)
```
**Old Output**
```python
[1/x, 1]
```

**New Output (since commit  8a06c07, release sympy-1.7rc1 and onwards)**
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
StopIteration
```

Is the new behaviour intentional and, if so, why is this preferable to the old behaviour? 
Is there a better way to compute a series one term at a time?

Thanks a lot for your help.

**Comment by oscarbenjamin** [comment]
Thanks for reporting this. I don't think that the change was intentional because this is a documented feature:
https://github.com/sympy/sympy/blob/425353e982439dc64f3d33876cce291aac628ed8/sympy/core/expr.py#L2806-L2810

Also this still works as long as the series isn't of a `Mul`:
```python
In [3]: from sympy import * 
   ...: x = Symbol('x') 
   ...: expr = (exp(x)).series(x,n=None) 
   ...: terms = [ next(expr) for i in range(10) ] 
   ...: print(terms)                                                                                                              
[1, x, x**2/2, x**3/6, x**4/24, x**5/120, x**6/720, x**7/5040, x**8/40320, x**9/362880]
```

This is a bug in both master and 1.7 and this is also a regression since 1.6.2.

If someone (@sachin-4099) can open a PR for this then I can make sure that the fix goes into the 1.7.1 bugfix release.

I was about to release 1.7.1 so this is now a release blocker.

**Comment by sachin-4099** [comment]
I will look into this and try to fix this as soon as possible.
